web/src/message_lists.ts

```diff
@@ -61,12 +62,26 @@ export function set_current(msg_list: MessageList | undefined): void {
 }
 
 export function update_current_message_list(msg_list: MessageList | undefined): void {
-    if (current && !current.preserve_rendered_state) {
+    if (current && !current.should_preserve_current_rendered_state()) {
         // Remove the current message list from the DOM.
         current.view.$list.remove();
         rendered_message_lists.delete(current.id);
     } else {
+        // We plan to keep the current message list cached.
         current?.view.$list.removeClass("focused-message-list");
+        // Remove any existing message lists that we have with the same filter.
+        // TODO: If we start supporting more messages lists than just Combined feed,
+        // make this a proper filter comparison between the lists.
+        if (current?.data.filter.is_in_home()) {
+            for (const [id, msg_list] of rendered_message_lists) {
+                if (id !== current.id && msg_list.data.filter.is_in_home()) {
+                    msg_list.view.$list.remove();
+                    rendered_message_lists.delete(id);
```

Comment:
Can you explain the consequences of removing it from `$list` and deleting it from `rendered_message_lists`? Will it get added back to that structure if visited again later?

Reply:
OK, read this more, and I understand `rendered_message_lists.delete`; I see that we'll potentially save this replaced message list for the combined feed. But not sure I understand the other half or what `msg_list.view.$list.remove();` does.

Reply:
```
    // Since we change `current` message list in the function, we need to decide if the
    // current message list needs to be cached or discarded. 
    // 
    // If we are caching the current message list, we need to remove any other message lists
    // that we have cached with the same filter. 
    //
    // If we are discarding the current message list, we need to remove the
    // current message list from the DOM.
```
    

Added this comment at the top of the function.

Reply:
OK, but here's what I don't understand: What happens when we visit the Combined Feed at some future point after doing that `rendered_message_lists.delete` operation? Does it just populate its `message_list_data` object from `all_messages_data` just like the very first time and end up in the same state as if we'd not done these deletions, or are there potential weirder consequences?

Reply:
So, in that case we have combined feed with `first_unread` cached as only delete we are preserving another list.

* If user just clicks on `combined feed` in left sidebar, we narrow to first unread.
* If user clicks browser back / `escape` key, we narrow with a `then_select_id`.  If cached `list.msg_id_in_fetched_range(opts.then_select_id)`, then we use the cached list, otherwise we try to populate a new list like any other. That would involve looking at `all_messages_data` to see if it has the `then_select_id`.